### ISSUE TITLE
Ability to generate complex types without setters/getters

### ISSUE THREAD

**Issue by fduch** [wsdl2phpgenerator/wsdl2phpgenerator#220]
First of all thank you for your work, your library is great!

I am trying to utilize wsdl2phpgenerator to generate client code to interact with some server classes shared via soap. On the server side i've got set of structs (ie DTO's - objects without method with only public properties) that are represented in wsdl-document like complex types and that i would like to reach on the client side exactly in the same manner (structs with public properties). When i try to generate client code based on server wsdl using wsdl2phpgenerator all the complex types becomes the look of the classes with protected properties bundled with getter/setter method for each one. 

Is there strong necessity in generation complex types filled with methods and protected members and may by it is possible to consider ability (perhaps as non-mandatory config option) to generate complex types with public properties instead of accessors? Technically it's not difficult and i could be glad to submit the PR provides this functionality.

**Comment by kasperg** [comment]
Thank for the feedback and your suggestion.

Currently I do not think public properties are worth pursuing. The problem is tied to handling of properties which are complex data types and how the PHP SoapClient processes data to and from the SOAP server.

When the PHP SoapClient takes an object as an argument to a method it expects the properties of the object to match the method regarding both their names and the format of their values. `wsdl2phpgenerator` currently supports converting primitives to objects for certain data types. This is currently only working for date/`DateTime` but I think it is something that will expand going forward - e.g. for Enums. The conversion between primitive and object currently takes place in getters, setters and constructors.

If `wsdl2phpgenerator` provided classes with public properties this conversion would have to take place elsewhere in the generated code. I believe this would increase the complexity of `wsdl2phpgenerator` significantly.

Even if we found somewhere else to place the conversion I still think having the unprocessed values around is helpful e.g. for debugging purposes.

I have to caveats:

1. Increasing complexity. Even if you would provide the pull request including unit tests and all I imagine it will increase the complexity of the code going forward and thus making the library harder to maintain. I do not think this increase in complexity is worth it for a feature which I consider cosmetic.
2. Having partly implemented configuration options. You might want to reduce the complexity of the implementation but not supporting complex properties. I do not think it is a good idea to add half-baked functionality.

If you still want to give the implementation a try I would be happy to take a look at it. Please remember my points above.

**Comment by fduch** [comment]
@kasperg i see your point: you prepare getters and setters in order to convert primitives into PHP representations and vice versa.

In this case i have one question more: what should i do when i have to do some custom type convertation without patching generated php code? For example if i want simply to use [Carbon](https://github.com/briannesbitt/Carbon) instead of DateTime or when i need to perform more complex type transformations (may be with some external code involved)? 
If we imagine that PHP's representation of complex type are flat (with public properties and without accessors) then such things can be done easily with external transformer code accesses structs directly or for example with the help of SoapClient's `typemap` option.

**Comment by kasperg** [comment]
@fduch: I agree that there may be a use case for transformation here: Using third party libraries such as Carbon, extending generated Complex types etc.

As said: If you are interested in taking a stab at implementing this or toss around some ideas I would be happy to take a look at it.
